checkbox-ng/checkbox_ng/launcher/subcommands.py

```diff
@@ -1333,6 +1333,48 @@ def __missing__(self, key):
                 print(job_id)
 
 
+class ListTestplan:
+    @property
+    def sa(self):
+        return self.ctx.sa
+
+    def register_arguments(self, parser):
+        parser.description = (
+            "Outputs the list of jobs and templates that will be run by a "
+            "given test plan as JSON. Instantiated jobs are ignored."
+        )
+        parser.add_argument("TEST_PLAN", help=_("test plan id to use"))
+
+    def invoked(self, ctx):
+        self.ctx = ctx
+        self.sa.start_new_session("checkbox-listing-ephemeral")
+        tps = self.sa.get_test_plans()
+        if ctx.args.TEST_PLAN not in tps:
+            raise SystemExit("Test plan not found")
+        self.sa.select_test_plan(ctx.args.TEST_PLAN)
+        self.sa.bootstrap()
+        job_and_template_list = []
```

Comment:
Minor small fix, use a set here, way faster to check if something is in there or not

Reply:
If I use a `set()` instead of a list `[]`, I get the following error:

```python
Traceback (most recent call last):
  File "/home/pieq/.venvs/checkbox/bin/checkbox-cli", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/pieq/dev/work/checkbox/checkbox-ng/checkbox_ng/launcher/checkbox_cli.py", line 165, in main
    return subcmd.invoked(ctx)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/pieq/dev/work/checkbox/checkbox-ng/checkbox_ng/launcher/subcommands.py", line 1375, in invoked
    job_and_template_set.add(attrs)
TypeError: unhashable type: 'dict'
```